core/src/test/scala/unit/kafka/admin/DescribeConsumerGroupTest.scala

```diff
@@ -315,24 +271,40 @@ class DescribeConsumerGroupTest extends KafkaServerTestHarness {
     val consumerGroupCommand = new KafkaConsumerGroupService(opts)
 
     TestUtils.waitUntilTrue(() => {
-        try {
           val (state, assignments) = consumerGroupCommand.describeGroup()
           state == Some("Stable") &&
           assignments.isDefined &&
           assignments.get.count(_.group == group) == 2 &&
           assignments.get.count{ x => x.group == group && x.partition.isDefined} == 2 &&
           assignments.get.count{ x => x.group == group && !x.partition.isDefined} == 0
-        } catch {
-          case _: GroupCoordinatorNotAvailableException | _: IllegalArgumentException =>
-            // Do nothing while the group initializes
-            false
-          case e: Throwable =>
-            throw e
-        }
       }, "Expected two rows (one row per consumer) in describe group results.")
 
     consumerGroupCommand.close()
   }
+
+  @Test
+  def testDescribeGroupWithNewConsumerWithShortInitializationTimeout() {
+    // run one consumer in the group consuming from a single-partition topic
+    val executor = new ConsumerGroupExecutor(brokerList, 1, group, topic)
+
+    // set the group initialization timeout too low for the group to stabilize
+    val cgcArgs = Array("--bootstrap-server", brokerList, "--describe", "--group", "group", "--timeout", "10")
+    val opts = new ConsumerGroupCommandOptions(cgcArgs)
+    val consumerGroupCommand = new KafkaConsumerGroupService(opts)
+
+    try {
+      val (state, assignments) = consumerGroupCommand.describeGroup()
+      fail("The consumer group command should fail due to low initialization timeout")
+    } catch {
+      case e @ (_: GroupCoordinatorNotAvailableException | _: IllegalArgumentException) =>
```

Comment:
Regarding the first point you raised, with a short 10 ms timeout used in the unit test `IllegalArgumentException` does not occur in my dev environment. But when I raise the timeout to 400 ms it raises that exception. Not sure what the best way to capture both in unit tests are because the proper timeout values (at least for `IllegalArgumentException`) for the unit test would probably be different from one environment to another. Any suggestion?

On your second point, do you mean when we reach the timeout we just throw a `TimeoutException` instead of continuing with the normal process (without any retries)? I guess it makes sense for user to see the impact or importance of the timeout value they can set. I'll try to update the PR with this. Please advise if I misunderstood your point.

Thanks.